src/components/show-products/index.js

```diff
@@ -16,9 +16,33 @@ export const ShowProducts = () => {
   const { products, loading } = useProducts();
   const { push } = useHistory();
   const { storedValue } = useLocalStorage(LOCAL_STORAGE_LIST_TOKEN);
-  const timeFrames = TimeFrames;
+  const [searchTerm, setSearchTerm] = useState('');
+  const [list, setList] = useState([]);
 
-  const handleOnChange = (productID) => {
+  useEffect(() => {
+    let results = products;
+
+    if (searchTerm !== '') {
+      results = products.filter(({ productName }) =>
+        productName
+          .toLowerCase()
+          .replace(/\s/g, '')
+          .normalize('NFD')
+          .replace(/[\u0300-\u036f]/g, '')
+          .includes(
+            searchTerm
+              .toLowerCase()
+              .replace(/\s/g, '')
+              .normalize('NFD')
+              .replace(/[\u0300-\u036f]/g, ''),
```

Comment:
I think you are comparing if two unicodes (accent and x). So smart!!!

However, I think it could be a little better if you can create extra constants and improve code readability and understand a little your magic logic. 😄  Also you can have a code reusable, because you can use those constants in multiple places.

P.S. Please, maybe you remember about ["Magic Numbers".](https://kyleshevlin.com/what-are-magic-values)

Reply:
Bonus: I believe you might create an isolate function maybe `normalizeWord`

```js
function normalizeWord(wordToNormalize){
    return wordToNormalize
          .toLowerCase()
          .replace(/\s/g, '')
          .normalize('NFD')
          .replace(/[\u0300-\u036f]/g, '');
}
```

If you will use some variables for unicodes ⭐ , I think It could be clearer and simpler.